addons/mail/static/src/core/web/messaging_menu_patch.js

```diff
@@ -178,4 +186,75 @@ patch(MessagingMenu.prototype, {
         }
         return _t("Failure: %(modelName)s", { modelName: failure.modelName });
     },
+    onClickThread(isMarkAsRead, thread) {
+        if (this.store.discuss.activeTab === "inbox") {
+            if (isMarkAsRead) {
+                thread.needactionMessages[0].setDone();
+                return;
+            } else {
+                thread.highlightMessage = thread.needactionMessages[0];
+            }
+        }
+        super.onClickThread(isMarkAsRead, thread);
+    },
+    get threads() {
+        if (this.store.discuss.activeTab === "inbox") {
+            return this._getInboxTabThreads();
+        }
+        return super.threads;
+    },
+    _getInboxTabThreads() {
+        const searchTerm = cleanTerm(this.store.discuss.searchTerm);
+        let allMessages = [
+            ...this.store.inbox.messages.reverse(),
+            ...this.store.history.messages.reverse(),
+        ];
+        if (searchTerm) {
+            allMessages = allMessages.filter(
+                (msg) =>
+                    cleanTerm(msg.thread.displayName).includes(searchTerm) ||
+                    cleanTerm(msg.previewText.toString()).includes(searchTerm)
+            );
+        }
+        return allMessages.map(
+            (msg, idx) =>
+                new Proxy(msg.thread, {
+                    get(target, prop) {
+                        const overrides = {
+                            localId: `${target.localId}-${msg.id || idx}`,
+                            needactionMessages: [msg],
+                            isUnread: msg.needaction ? target.isUnread : false,
+                            needactionCounter: 0,
+                        };
+                        return prop in overrides ? overrides[prop] : target[prop];
+                    },
+                })
+        );
+    },
+    get notificationItems() {
+        return Array.from(super.notificationItems).filter((el) =>
+            el.classList.contains("o-mail-NotificationItem")
+        );
+    },
+    get isInboxTabLoading() {
+        return this.store.inbox.status === "loading" || this.store.history.status === "loading";
+    },
+    get hasInboxTabLoadingFailed() {
+        return this.store.inbox.hasLoadingFailed || this.store.history.hasLoadingFailed;
+    },
+    async loadMoreInboxTabData() {
+        if (this.store.inbox.loadOlder) {
+            await this.store.inbox.fetchMoreMessages();
+            if (this.threads.length < 15) {
+                await this.loadMoreInboxTabData();
+            }
+            return;
+        }
+        if (this.store.history.loadOlder) {
+            await this.store.history.fetchMoreMessages();
+            if (this.threads.length < 15) {
+                await this.loadMoreInboxTabData();
+            }
+        }
+    },
```

Comment:
Load more should load inbox fully before considering loading history at all. If you load both at the same time, you introduce holes and this is a problem. Even if you could load it, the loader is by reaching the bottom (see only history items) and some inbox items would pop somewhere at the top out of the viewport, which is very poor UX.